client/sniffer/src/main/java/org/elasticsearch/client/sniff/Sniffer.java

```diff
@@ -116,15 +118,19 @@ void sniffOnFailure(HttpHost failedHost) {
         void sniff(HttpHost excludeHost, long nextSniffDelayMillis) {
             if (running.compareAndSet(false, true)) {
                 try {
-                    List<HttpHost> sniffedHosts = hostsSniffer.sniffHosts();
+                    final List<Node> sniffedHosts = hostsSniffer.sniffHosts();
                     logger.debug("sniffed hosts: " + sniffedHosts);
                     if (excludeHost != null) {
-                        sniffedHosts.remove(excludeHost);
+                        for (Iterator<Node> itr = sniffedHosts.iterator(); itr.hasNext();) {
```

Comment:
should we update the failure listener to get a Node as an argument rather than a host? I think that would be the right thing to do but it's going to break users who have implemented a failure listener. We should document this as a breaking change if we do so.

Reply:
There are so many breaking changes in this! I think I'll need to make a list so we can decide which are OK.

Reply:
I can update the interface but I didn't want to do it now because it is breaking. Maybe I can do that in master only? Is that worth doing?

Reply:
I think changing the failure listener would make things consistent, yet it is the least important out of all these breaking changes. Yet, I think that breaking performRequest methods is one thing, pretty bad, while breaking extensions points will bother less users, who can easily figure out what's going on thanks ton the compiler.

Reply:
This change doesn't break `performRequest` though. It breaks *mocking* `performRequest` which is *weird*.

Reply:
sure I didn't mean that you are breaking performRequest, I meant to say that what we are breaking is less breaking than if we would have broken performRequest :)